docs/upgrade.html

```diff
@@ -21,6 +21,10 @@
 0.10.0.0 has <a href="#upgrade_10_breaking">potential breaking changes</a> (please review before upgrading) and
 there may be a <a href="#upgrade_10_performance_impact">performance impact during the upgrade</a>. Because new protocols
 are introduced, it is important to upgrade your Kafka clusters before upgrading your clients.
+<p/>
+<b>Notes to client with version 0.9.0.0: </b>Due to a bug introduced in 0.9.0.0,
+clients with ZooKeeper dependency may not be able to work with 0.10.0.0 broker. Therefore, 0.9.0.0 clients should
```

Comment:
@junrao In future releases we expect to remove the old Scala clients. This means only the brokers will talk directly to Zookeeper and this should not longer be an issue. My understanding is the goal of 0.10.0 is to be a "compatible" release. Future releases may remove other deprecated things and choose to be breaking. I think that's the best time to break here as well. 

I don't want to push the issue too much. In the end, I am okay with either choice. I just want to be sure we are consciously choosing to break and for good reason.